iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergBranchExec.java

```diff
@@ -49,9 +47,13 @@ public static void createBranch(Table table, AlterTableSnapshotRefSpec.CreateSna
     } else if (createBranchSpec.getAsOfTime() != null) {
       snapshotId = SnapshotUtil.snapshotIdAsOfTime(table, createBranchSpec.getAsOfTime());
     } else if (createBranchSpec.getAsOfTag() != null) {
-      Snapshot snapshot = Optional.ofNullable(table.snapshot(createBranchSpec.getAsOfTag())).orElseThrow(() ->
-          new IllegalArgumentException(String.format("Tag %s does not exist", createBranchSpec.getAsOfTag())));
-      snapshotId = snapshot.snapshotId();
+      String tagName = createBranchSpec.getAsOfTag();
+      SnapshotRef snapshotRef = table.refs().get(tagName);
+      if (snapshotRef != null && snapshotRef.isTag()) {
```

Comment:
Sorry for my carelessness. We need to check the `TAG` type before getting its snapshotId.
cc @ayushtkn